glean-core/rlb/src/private/object.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::marker::PhantomData;
+
+use glean_core::traits;
+
+use crate::ErrorType;
+
+// We need to wrap the glean-core type: otherwise if we try to implement
+// the trait for the metric in `glean_core::metrics` we hit error[E0117]:
+// only traits defined in the current crate can be implemented for arbitrary
+// types.
+
+/// Developer-facing API for recording event metrics.
+///
+/// Instances of this class type are automatically generated by the parsers
+/// at build time, allowing developers to record values that were previously
+/// registered in the metrics.yaml file.
+#[derive(Clone)]
+pub struct ObjectMetric<K> {
+    pub(crate) inner: glean_core::metrics::ObjectMetric,
+    object_type: PhantomData<K>,
```

Comment:
...remind me what this is doing?

Reply:
Thanks. The Nomicon on PhantomData is not real helpful in explaining this. ...but now, of course, after I scroll a bit, I find [the docs aren't unhelpful](https://doc.rust-lang.org/std/marker/struct.PhantomData.html#unused-type-parameters)